// pages/add-product/index.js
import React, {useState, useEffect} from 'react'

import useFetch from 'hooks/useFetch'

const AddProduct = () => {
    const apiUrl = '/create-food'
    const [{isLoading, error, response}, doFetch] = useFetch(apiUrl)
    const [nameProduct, setNameProduct] = useState('')
    const [animalProtein, setAnimalProtein] = useState(0)
    const [vegetableProtein, setVegetableProtein] = useState(0)
    const [animalFats, setAnimalFats] = useState(0)
    const [vegetableFats, setVegetableFats] = useState(0)
    const [fastCarbohydrates, setFastCarbohydrates] = useState(0)
    const [slowCarbohydrates, setSlowCarbohydrates] = useState(0)
    const [calories, setCalories] = useState(0)

    const handleSubmit = event => {
        event.preventDefault()

        const data = {
            name: nameProduct,
            protein_a: animalProtein,
            protein_v: vegetableProtein,
            fats_a: animalFats,
            fats_v: vegetableFats,
            carbohydrates_f: fastCarbohydrates,
            carbohydrates_s: slowCarbohydrates,
            calories: calories
        }

        doFetch({
            method: 'post',
            data
        })
    }

    useEffect(() => {
        if (!response) {
            return
        }

    }, [response])

    return (
        <div className="container">

            <div className="row">
                <div className="col">
                    <div className="col-md-6 offset-md-3 col-xs-12">
                        <form  onSubmit={handleSubmit}>
                            <fieldset>
                                <fieldset className="form-group">
                                    <input
                                        type="text"
                                        className="form-control"
                                        placeholder="Name of product"
                                        value={nameProduct}
                                        onChange={e => setNameProduct(e.target.value)}
                                    />
                                </fieldset>
                                <fieldset className="form-group">
                                    <input
                                        type="text"
                                        className="form-control"
                                        placeholder="Animal protein"
                                        aria-label="Animal protein"
                                        value={animalProtein}
                                        onChange={e => setAnimalProtein(e.target.value)}
                                    />
                                </fieldset>
                                <fieldset className="form-group">
                                    <input
                                        type="text"
                                        className="form-control"
                                        placeholder="Vegetable protein"
                                        aria-label="Vegetable protein"
                                        value={vegetableProtein}
                                        onChange={e => setVegetableProtein(e.target.value)}
                                    />
                                </fieldset>
                                <fieldset className="form-group">
                                    <input
                                        type="text"
                                        className="form-control"
                                        placeholder="Animal fat"
                                        aria-label="Animal fat"
                                        value={animalFats}
                                        onChange={e => setAnimalFats(e.target.value)}
                                    />
                                </fieldset>
                                <fieldset className="form-group">
                                    <input
                                        type="text"
                                        className="form-control"
                                        placeholder="Vegetable fat"
                                        aria-label="Vegetable fat"
                                        value={vegetableFats}
                                        onChange={e => setVegetableFats(e.target.value)}
                                    />
                                </fieldset>
                                <fieldset className="form-group">
                                    <input
                                        type="text"
                                        className="form-control"
                                        placeholder="Fast carbohydrates"
                                        aria-label="Fast carbohydrates"
                                        value={fastCarbohydrates}
                                        onChange={e => setFastCarbohydrates(e.target.value)}
                                    />
                                </fieldset>
                                <fieldset className="form-group">
                                    <input
                                        type="text"
                                        className="form-control"
                                        placeholder="Slow carbohydrates"
                                        aria-label="Slow carbohydrates"
                                        value={slowCarbohydrates}
                                        onChange={e => setSlowCarbohydrates(e.target.value)}
                                    />
                                </fieldset>
                                <fieldset className="form-group">
                                    <input
                                        type="text"
                                        className="form-control"
                                        placeholder="Calories"
                                        aria-label="Calories"
                                        value={calories}
                                        onChange={e => setCalories(e.target.value)}
                                    />
                                </fieldset>
                                <button
                                    className="btn btn-lg btn-success"
                                    type="submit"
                                >
                                    Button
                                </button>
                            </fieldset>
                        </form>
                    </div>
                </div>
            </div>

        </div>
    )
}

export default AddProduct
